setup.py

```diff
@@ -131,11 +133,83 @@ def make_cython_ext(name, module, sources):
     return extension
 
 
-def get_requirements(filename='requirements.txt'):
-    here = os.path.dirname(os.path.realpath(__file__))
-    with open(os.path.join(here, filename), 'r') as f:
-        requires = [line.replace('\n', '') for line in f.readlines()]
-    return requires
+def parse_requirements(fname='requirements.txt', with_version=False):
```

Comment:
Better to set the default value of `with_version` to True?

Reply:
I've generally found that its best to not specify a version in the `install_requires`. This is mostly due to pip's limited ability to resolve dependencies. I've often found that it causes more problems than it solves. However, doing a `pip install -r requirements.txt` will respect the version requirements. 

That being said, only a few requirements actually have version dependencies in the txt files. Also it looked like the old parsing script did include the versions, so perhaps its ok to include them in this instance. I've enabled it to ensure that the existing behavior is preserved and we can always disable it if there are any problems.